pyproject.toml

```diff
@@ -30,9 +30,9 @@ authors = [
 license = {file = 'LICENSE'}
 requires-python = '>=3.9'
 dependencies = [
-  "numpy>=1.22.4,<2; python_version<'3.11'",
-  "numpy>=1.23.2,<2; python_version=='3.11'",
-  "numpy>=1.26.0,<2; python_version>='3.12'",
+  "numpy>=1.22.4; python_version<'3.11'",
+  "numpy>=1.23.2; python_version=='3.11'",
+  "numpy>=1.26.0; python_version>='3.12'",
```

Comment:
This PR removed the numpy<2 pin again, which is good for the _development_ branch (and nightly wheels), but shouldn't we still keep the pin for the _release_ branch? 

Unless we are planning to build the pandas 2.2.0 wheels with nightly numpy (which isn't the case right now), the upcoming pandas 2.2.0 release is still not ABI compatible with numpy 2.0

Reply:
I think this is right by my understanding of packaging around the NumPy 2 release.  Should keep the pin on until wheels can be built against NumPy 2.0.0rc1. Then can build against NumPy>=2.0.0rc1 and can have runtime dependency of NumPy 1.2x+.

Reply:
Slightly off topic but do we know what prevents us from using build isolation with meson? That would seemingly make this issue easier to manage as well

Reply:
> Slightly off topic but do we know what prevents us from using build isolation with meson? That would seemingly make this issue easier to manage as well

Sorry, not quite following here. 
IIUC, build isolation means building the package in a separate environment from the one where it's installed in.

We currently just disable it for editable builds (otherwise the auto-rebuilding wouldn't work), and also on regular CI (since there's no point in installing the build tools again when they're already installed in the conda environment).

I'm pretty sure cibuildwheel might use it, though.

Reply:
> since there's no point in installing the build tools again when they're already installed in the conda environment)

The benefit depends on if you consider the conda environment to be a build environment or a runtime environment. The advantage of an isolated build environment is you can build your project against any particular version like `oldest-supported-numpy` for binary compatability but choose a different version for the runtime.

I haven't closely tracked the particular case of NumPy 2.0 but it sounds like that could be useful based off of @bashtage comment https://github.com/pandas-dev/pandas/pull/55739#discussion_r1446081417